src/components/search/SearchDirective.js

```diff
@@ -149,40 +153,31 @@ goog.require('ga_search_type_directives');
             // Standard query then
             var url = gaUrlUtils.append($scope.options.searchUrl,
                                         'searchText=' + encodeURIComponent(q));
-            url = gaUrlUtils.append(url, 'lang=' + $translate.use());
-            url = $scope.options.applyTopicToUrl(url,
-                $scope.childoptions.currentTopic);
+            url = gaUrlUtils.append(url, 'lang=' + gaLang.get());
+            url = url.replace('{Topic}', gaTopic.get().id);
 
             $scope.childoptions.baseUrl = url;
             $scope.childoptions.query = q;
           }
         } else {
           blockQuery = false;
+          $scope.$broadcast('gaSearchReset');
```

Comment:
No. It shouldn't be a problem. In main directive, it's reacting to changes and will reset child query to actual value.

But it's maybe not ideal as it is right now.